contracts/contracts/ocr2/SequencerUptimeFeed/library.cairo

```diff
@@ -0,0 +1,258 @@
+
+    return ()
+end
+
+# TODO: this methods are virtual in .sol: clarify
+@external
+func set_l1_sender{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    address : felt
+):
+    Ownable_only_owner()
+    _set_l1_sender(address)
+
+    return ()
+end
+
+@view
+func l1_sender{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}() -> (
+    address : felt
+):
+    let (address) = s_l1_sender.read()
+    return (address)
+end
+
+namespace sequencer_uptime_feed:
+    func constructor{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        initial_status : felt, owner_address : felt
+    ):
+        assert_boolean(initial_status)
+
+        simple_read_access_controller.constructor(owner_address)
+
+        let (timestamp) = get_block_timestamp()
+        _record_round(1, initial_status, timestamp)
+
+        return ()
+    end
+
+    func update_status{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        from_address : felt, status : felt, timestamp : felt
+    ):
+        alloc_locals
+        require_l1_sender(from_address)
+        assert_boolean(status)
+
+        let (latest_round_id) = s_latest_round_id.read()
+        let (latest_observation_timestamp) = s_rounds.read(
+            latest_round_id, Round.observation_timestamp
+        )
+        let (local latest_status) = s_rounds.read(latest_round_id, Round.answer)
+
+        let (lt) = is_le(timestamp, latest_observation_timestamp - 1)  # timestamp < latest_observation_timestamp
+        if lt == TRUE:
+            UpdateIgnored.emit(latest_status, latest_observation_timestamp, status, timestamp)
+            return ()
+        end
+
+        if latest_status == status:
+            _update_round(latest_round_id, status)
+        else:
+            let latest_round_id = latest_round_id + 1
+            _record_round(latest_round_id, status, timestamp)
+        end
+
+        return ()
+    end
+
+    func latest_round_data{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}() -> (
+        round : Round
+    ):
+        require_access()
+
+        let (lateset_round_id) = s_latest_round_id.read()
+        let (latest_round) = sequencer_uptime_feed.round_data(lateset_round_id)
+
+        return (latest_round)
+    end
+
+    func round_data{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        round_id : felt
+    ) -> (res : Round):
+        require_access()
+        require_valid_round_id(round_id)
+
+        let (round) = _get_round(round_id)
+        return (round)
+    end
+
+    func description{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}() -> (
+        description : felt
+    ):
+        const description = 'L2 Sequencer Uptime Status Feed'
+        return (description)
+    end
+
+    func decimals{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}() -> (
+        decimals : felt
+    ):
+        const decimals = 0
```

Comment:
same here.  Can we just do `return 0` or is this not possible with Cairo?

Reply:
Done